data/templates/frr/isis.frr.tmpl

```diff
@@ -31,6 +31,67 @@ router isis {{ process }}
 {% if spf_interval is defined and spf_interval is not none %}
  spf-interval {{ spf_interval }}
 {% endif %}
+{% if traffic_engineering is defined %}
```

Comment:
Please use: `{% if traffic_engineering is defined and traffic_engineering is not none %}`, thus if TE is not enabled the entire tree won't be run which is faster.

The difference is that if a dict contains an empty `traffic_engineering` key this evaluates to true, but if there are no more sub-keys, it evaluates to false entirely at once.

Reply:
Ahh yes, that makes sense then. Will make said changes.